### PR TITLE
Add asdf version instructions

### PR DESCRIPTION
**Pull request by daxmc99**
Help internal developers onboard onto Terraform faster

### CODE REVIEW

File: doc/dev/background-information/languages/terraform.md [hunk 1]
```diff
@@ -3,6 +3,10 @@
 - General Terraform [styleguide](https://www.terraform.io/docs/configuration/style.html)
 - Sourcegraph Terraform [Extended Guide](./extended_guide/terraform.md)
 
+## Terraform versioning
+
+We use [asdf](https://asdf-vm.com/#/) to manage Terraform versions. After installing Asdf into your shell, run `asdf install && asdf reshim` to pickup the Terraform version defined in the `.tool-version` file for the specific Terraform module.
```
Comment (ggilmore) [inline doc/dev/background-information/languages/terraform.md:8]: Why is reshim necessary?

Comment (daxmc99) [reply doc/dev/background-information/languages/terraform.md:8]: Depending on shell env I have seen this allow Asdf pickup the correct version without needing to restart the shell

Comment (daxmc99) [reply doc/dev/background-information/languages/terraform.md:8]: *if you reshim

**Review by ggilmore — APPROVED** [review]
See also: 

- https://github.com/sourcegraph/infrastructure/blob/176c55a5de93bf953b2acef327314be3f0b95cbf/justfile#L39-L41
- https://github.com/sourcegraph/infrastructure/blob/176c55a5de93bf953b2acef327314be3f0b95cbf/scripts/terraform-init.sh#L1
- https://github.com/sourcegraph/infrastructure/blob/176c55a5de93bf953b2acef327314be3f0b95cbf/scripts/asdf-install-all.sh#L1


(These are the exact scripts that infrastructure's CI uses).

**Pull request merged**